src/Core/OrganizationFeatures/OrganizationLicenses/UpdateOrganizationLicenseCommand.cs

```diff
@@ -0,0 +1,144 @@
+﻿#nullable enable
+
+using System.Text.Json;
+using Bit.Core.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Exceptions;
+using Bit.Core.Models.Business;
+using Bit.Core.Models.OrganizationConnectionConfigs;
+using Bit.Core.OrganizationFeatures.OrganizationLicenses.Interfaces;
+using Bit.Core.Models.Data.Organizations;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Bit.Core.Utilities;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationLicenses;
+
+public class UpdateOrganizationLicenseCommand : IUpdateOrganizationLicenseCommand
+{
+    private readonly ILicensingService _licensingService;
+    private readonly IGlobalSettings _globalSettings;
+    private readonly IOrganizationService _organizationService;
+
+    public UpdateOrganizationLicenseCommand(
+        ILicensingService licensingService,
+        IGlobalSettings globalSettings,
+        IOrganizationService organizationService)
+    {
+        _licensingService = licensingService;
+        _globalSettings = globalSettings;
+        _organizationService = organizationService;
+    }
+
+    public async Task UpdateLicenseAsync(Organization organization, OrganizationLicense license, 
+        Organization? existingOrganization, OrganizationPlanUsage planUsage)
+    {
+        license.CanUse(_globalSettings, _licensingService);
+        ValidateLicenseForOrganizationAsync(organization, license, existingOrganization, planUsage);
+        
+        await WriteLicenseFileAsync(organization, license);
+        await UpdateOrganizationAsync(organization, license);
+    }
+
+    private void ValidateLicenseForOrganizationAsync(Organization organization, OrganizationLicense license, 
```

Comment:
These validations are difficult to test as written.

It would be preferable to invent a new `Organization` model, `SelfHostedOrganizationView` or something that combines `Organization` and `OrganizationPlanUsage`. This model would be capable of validating if a license is applicable to the current self-hosted usage.